src/v2/guide/migration-vue-router.md

```diff
@@ -236,6 +236,20 @@ if (route.meta.requiresAuth) {
 </div>
 {% endraw %}
 
+### [] Syntax for Array of Route Query <sup>removed</sup>
+
+In URL, [] syntax for array of query is removed. For example, `/foo?users[]=Tom&users[]=Jerry` will be `/foo?users=Tom&users=Jerry` now.
+
+Since vue-router will treat `?users=Tom` as `query.users == 'Tom'`, unless you are for sure that the array will have more than one element, you should add the following kind of watcher:
+```javascript
+'$route.query.users': {
+  handler(val) {
+    this.$route.query.users = Array.isArray(val) ? val : [val]
```

Comment:
Why should we talk about watchers? I think the relevant part is that the user should check if `val` is an array, right?